src/transformState.js

```diff
@@ -4,8 +4,41 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        removeProperties(state, action.keysToRemove);
+
+        break;
+
+      case 'clear':
+        clear(state);
+
+        break;
+
+      default:
+        throw new Error(`${action.type} is not defined !`);
+    }
+  }
+}
+
+function removeProperties(object, params) {
```

Comment:
The function name is `removeProperties` so better name for second argument would be `propertiesToRemove` or `keysToRemove`